bashlex/parser.py

```diff
@@ -607,6 +607,8 @@ def parse(s, strictmode=True, expansionlimit=None, convertpos=False):
     expansionlimit is used to limit the amount of recursive parsing done due to
     command substitutions found during word expansion.
     '''
+    # remove starting and ending newlines and spaces
+    s = s.strip()
```

Comment:
what does this do to start/end positions of tokens? shouldn't we adjust those if we stripped?